### PR TITLE
Migrate config to class and provide API endpoints

### PR DESCRIPTION
**Pull request by HazardCreative**
Disassociate server-scoped config from event-scoped data.
- simplifies archiving and swapping events
- makes system function along a more typical paradigm
- provides for a future UI-based configurator, eliminating the need to edit config.json directly

### CODE REVIEW

File: src/interface/serial_node.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def read_firmware_timestamp(self):
 
 def discover(idxOffset, config, isS32BPillFlag=False, *args, **kwargs):
     nodes = []
-    config_ser_ports = getattr(config, 'SERIAL_PORTS', [])
+    config_ser_ports = config.get_item('GENERAL', 'SERIAL_PORTS')
```
Comment (ethomas997) [inline src/interface/serial_node.py:232]: This no longer has the empty-list default value, so will break if 'SERIAL_PORTS' is not present.

File: src/server/RaceContext.py [hunk 2]
```diff
@@ -33,6 +33,7 @@ def __init__(self):
         self.raceclass_rank_manager = None
         self.race_points_manager = None
 
+        self.serverconfig = None
```
Comment (ethomas997) [inline src/server/RaceContext.py:36]: Would be good to set this to its initial value

File: src/interface/serial_node.py (hunk 1 above)
Comment (HazardCreative) [reply src/interface/serial_node.py:232]: SERIAL_PORTS is always present in the internal default configuration that `config.json` applies on top of..

Comment (ethomas997) [reply src/interface/serial_node.py:232]: Still better to have a default value; there's no guarantee of that assumption at this point in the code

File: src/server/Config.py [hunk 3]
```diff
@@ -1,131 +1,256 @@
...
+
+        self.config = {
+            'SECRETS': {},
+            'GENERAL': {},
+            'HARDWARE': {},
+            'LED': {},
+            'LOGGING': {},
+            'SENSORS': {},
+        }
+
```
Comment (i-am-grub) [inline src/server/Config.py:24]: Defaults for `ADMIN_USERNAME` and `ADMIN_PASSWORD`

 `ADMIN_USERNAME` and `ADMIN_PASSWORD` are currently only generated in the new config when migrating from the legacy config. New installations cannot log into the timer without manually adding the parameters to the config file.

**Review by i-am-grub — CHANGES_REQUESTED** [review]
- Populate defaults for server credentials
- Update the use of `Config.GENERAL` in RHRace.py

**Review by i-am-grub — APPROVED** [review]
Speed ran through testing a few things and didn't come across anything that stands out as an issue.

The UI does act a looks weird when using the same database after upgrading to this PR (since the color values are now stored in the config), but if the user is following the recommend installation/upgrade documentation, this won't be an issue.

**Pull request merged**